File: add/codesnippet/CSharp/88c77b70-6ab7-4d6b-914d-_1.cs
//Define users and roles.
PrincipalPermission ppBob = new PrincipalPermission("Bob", "Manager");
PrincipalPermission ppLouise = new PrincipalPermission("Louise", "Supervisor");
PrincipalPermission ppGreg = new PrincipalPermission("Greg", "Employee");

//Define groups of users.
PrincipalPermission pp1 = (PrincipalPermission)ppBob.Union(ppLouise);
PrincipalPermission pp2 = (PrincipalPermission)ppGreg.Union(pp1);

File: add/codesnippet/CPP/88c77b70-6ab7-4d6b-914d-_1.cpp
      //Define users and roles.
      PrincipalPermission^ ppBob = gcnew PrincipalPermission( "Bob", "Manager" );
      PrincipalPermission^ ppLouise = gcnew PrincipalPermission( "Louise", "Supervisor" );
      PrincipalPermission^ ppGreg = gcnew PrincipalPermission( "Greg", "Employee" );
      
      //Define groups of users.
      PrincipalPermission^ pp1 = (PrincipalPermission^) (ppBob->Union( ppLouise ));
      PrincipalPermission^ pp2 = (PrincipalPermission^) (ppGreg->Union( pp1 ));